api-server/server/boot/donate.js

```diff
@@ -243,24 +255,73 @@ export default function donateBoot(app, done) {
           .send({ error: 'Donation failed due to a server error.' })
       );
   }
+  function addDonation(req, res) {
+    const { user, body } = req;
 
-  const pubKey = keys.stripe.public;
+    if (!user || !body) {
+      return res
+        .status(500)
+        .send({ error: 'User must be signed in for this request.' });
+    }
+    return Promise.resolve(req)
+      .then(
+        user.updateAttributes({
+          isDonating: true
+        })
+      )
+      .then(() => res.status(200).json({ isDonating: true }))
+      .catch(err => {
+        log(err.message);
+        return res.status(500).send({
+          type: 'danger',
+          message: 'Something went wrong.'
+        });
+      });
+  }
+
+  function updatePaypal(req, res) {
+    const { headers, body } = req;
+    return Promise.resolve(req)
+      .then(verifyWebHookType)
+      .then(getAsyncPaypalToken)
+      .then(token => verifyWebHook(headers, body, token, paypalWebhookId))
+      .then(hookBody => updateUser(hookBody, app))
+      .then(() => res.status(200).json({ message: 'received hook' }))
+      .catch(err => {
+        log(err.message);
+        return res.status(200).json({ message: 'received hook' });
+      });
+  }
+
+  const stripeKey = keys.stripe.public;
   const secKey = keys.stripe.secret;
+  const paypalKey = keys.paypal.client;
+  const paypalSec = keys.paypal.secret;
   const hmacKey = keys.servicebot.hmacKey;
-  const secretInvalid = !secKey || secKey === 'sk_from_stripe_dashboard';
-  const publicInvalid = !pubKey || pubKey === 'pk_from_stripe_dashboard';
+  const stripeSecretInvalid = !secKey || secKey === 'sk_from_stripe_dashboard';
+  const stripPublicInvalid =
+    !stripeKey || stripeKey === 'pk_from_stripe_dashboard';
+
+  const paypalSecretInvalid =
+    !paypalKey || paypalKey === 'id_from_paypal_dashboard';
+  const paypalPublicInvalid =
+    !paypalSec || paypalSec === 'secret_from_paypal_dashboard';
   const hmacKeyInvalid =
     !hmacKey || hmacKey === 'secret_key_from_servicebot_dashboard';
+  const paypalInvalid = paypalPublicInvalid || paypalSecretInvalid;
+  const stripeInvalid = stripeSecretInvalid || stripPublicInvalid;
 
-  if (secretInvalid || publicInvalid || hmacKeyInvalid) {
+  if (stripeInvalid || paypalInvalid || hmacKeyInvalid) {
     if (process.env.FREECODECAMP_NODE_ENV === 'production') {
-      throw new Error('Stripe API keys are required to boot the server!');
+      throw new Error('Donation API keys are required to boot the server!');
     }
-    console.info('No Stripe API keys were found, moving on...');
+    log('Some Donation API keys were not found, moving on...');
```

Comment:
```suggestion
    log('Donation disabled in development unless ALL test keys are provided');
```
Just to remind people they can't test, say, PayPal with only those keys.